src/main/java/org/kitteh/irc/client/library/element/DCCExchange.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * * Copyright (C) 2013-2016 Matt Baxter http://kitteh.org
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.element;
+
+import java.net.SocketAddress;
+import java.util.Optional;
+
+/**
+ * Represents an exchange using DCC.
+ */
+public interface DCCExchange extends Actor {
+    /**
+     * Gets the socket address of the local end.
+     * May return {@link Optional#empty()} if not connected.
+     *
+     * @return the socket address of the local end
+     */
+    Optional<SocketAddress> getLocalSocketAddress();
+
+    /**
+     * Gets the socket address of the remote end.
+     * May return {@link Optional#empty()} if not connected.
+     *
+     * @return the socket address of the remote end
+     */
+    Optional<SocketAddress> getRemoteSocketAddress();
+
+    /**
+     * Gets the connection status.
+     *
+     * @return {@code true} if the exchange is connected, otherwise false
```

Comment:
I should probably update my stuff to be `{@code true}` but why do you not do it for false?